### PR TITLE
Circleci project setup

### PR DESCRIPTION
**Pull request by carelynntsai**
## Overview

[Notion task](https://www.notion.so/uwblueprintexecs/Continuous-Integration-Setup-fe224cc6dd614b629300c2c6ef89f1ab)

Adds circleci integration

## Changes
- Adds config file for circleci integration
- To run docker commands, had to add them to Dockerfile.dev and changer docker version on docker-compose-dev.yml file

## Screenshots
image

## Testing
- build tests (runs docker commands, checks if the container is built, if docker is installed, runs npm install)

## Checklist
- [x] Change title to be more descriptive (shouldn't be just branch name)
- [x] Update `Notion task` link
- [x] `Overview`
- [x] `Changes`
  - [ ] \(Optional) Add inline comments in diff
- [x] \(Optional) `Screenshots`
- [x] \(Optional) `Testing` (N/A if nothing changed)
- [x] \(Optional) `Other Notes`
- [ ] Request at least 2 reviewers
- [x] Assign yourself to the PR
- [x] Make sure build passes

@uwblueprint/paramedics

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            - v1-dependencies-
+      - run: yarn test
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
```
Comment (petorsong) [inline .circleci/config.yml:71]: let's take this out

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  
+workflows:
+  version: 2
+  build-test:
+    jobs:
+      - install
+      - build:
+          requires:
+            - install
+      # - test:
```
Comment (petorsong) [inline .circleci/config.yml:81]: this out as well

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          name: Run Docker compose
+          command: |
+            set -x
+            docker-compose -f docker-compose-dev.yaml up -d
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+  
+  test:
```
Comment (petorsong) [inline .circleci/config.yml:52]: is this test section worth keeping in? we can always put it back in once the testing stuff also gets merged in

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+          docker_layer_caching: false
+      - run:
+          name: Install docker-compose
+          command: |
+            set -x
+            curl -L https://github.com/docker/compose/releases/download/1.12.0/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
+            chmod +x /usr/local/bin/docker-compose
+      - run:
+          name: Building image
+          command: docker build -t aaronabraham311/paramedics-web-docker:0.0.2 -f Dockerfile.dev .
```
Comment (JasonYG) [inline .circleci/config.yml:41]: I'm not too sure how this all works, but should `aaronabraham311` be here?

Comment (aaronabraham311) [reply .circleci/config.yml:41]: Good catch I'll fix that real quick

**Pull request merged**